packages/common/src/utils/formatting.ts

```diff
@@ -282,28 +277,29 @@ export function formatFieldValue(
         case DimensionType.DATE:
         case MetricType.DATE:
             return isMomentInput(value)
-                ? formatDate(
-                      value,
-                      isDimension(field) ? field.timeInterval : undefined,
-                      convertToUTC,
-                  )
+                ? formatDate(value, field.timeInterval, convertToUTC)
                 : 'NaT';
         case DimensionType.TIMESTAMP:
             return isMomentInput(value)
-                ? formatTimestamp(
-                      value,
-                      isDimension(field) ? field.timeInterval : undefined,
-                      convertToUTC,
-                  )
+                ? formatTimestamp(value, field.timeInterval, convertToUTC)
                 : 'NaT';
         case MetricType.MAX:
         case MetricType.MIN: {
             if (value instanceof Date) {
-                return formatTimestamp(
-                    value,
-                    isDimension(field) ? field.timeInterval : undefined,
-                    convertToUTC,
-                );
+                if (
+                    field.timeInterval &&
+                    [
+                        TimeFrames.YEAR,
+                        TimeFrames.QUARTER,
+                        TimeFrames.MONTH,
+                        TimeFrames.WEEK,
+                        TimeFrames.DAY,
+                    ].includes(field.timeInterval)
+                ) {
+                    return formatDate(value, field.timeInterval, convertToUTC);
+                }
+
+                return formatTimestamp(value, field.timeInterval, convertToUTC);
```

Comment:
maybe it will be better to create a new formatting function which will cover both time and date?